### PR TITLE
Design story final prod

### PR DESCRIPTION
**Pull request by rounak610**
### Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs update

### Checklist
- [ ] My pull request is atomic and focuses on a single change.
- [ ] I have documented my changes clearly and comprehensively.
- [ ] I have added the required tests.

### CODE REVIEW

**Review by czezi — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -2,6 +2,15 @@ FROM public.ecr.aws/docker/library/golang:1.22.3-bookworm AS build-base
 
 RUN apt-get update && apt-get install -y jq postgresql-client && apt-get clean && rm -rf /var/lib/apt/lists/*
 
+RUN useradd -m coder
```
[the remark this answers is not in the document]
Comment (rounak610) [reply Dockerfile:5]: The commit and raise pr functionality runs GitHub commands inside the server, so if we do not create user "coder" then they are done by "root" and when we run a backend story inside python executor, the GitHub commands are run by "coder", this results in dubious ownership and insufficient permissions errors;

File: Dockerfile [hunk 2]
```diff
@@ -65,6 +77,11 @@ ENTRYPOINT ["bash", "-c", "/entrypoint.d/initialise.sh && /go/executor"]
 
 FROM superagidev/supercoder-node-ide:latest AS node-executor
 
+USER coder
```
[the remark this answers is not in the document]
Comment (rounak610) [reply Dockerfile:80]: when the dubious ownership and insufficient permissions errors occurred in staging rakesh added this user in node executor also

File: Jenkinsfile
[the remark this answers is not in the document]
Comment (rounak610) [reply Jenkinsfile:1]: fixed

**Review by czezi — CHANGES_REQUESTED** [review]

File: workspace-service/app/config/frontend_worskpace_path.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+	config *koanf.Koanf
+}
+
+func (c *FrontendWorkspacePathConfig) FrontendWorkspacePath(projectHashID string, storyHashId string) string{
+	output := filepath.Join("/workspaces", projectHashID, "frontend" , ".stories" , storyHashId)
+	fmt.Println("___frontend workspace service____",output)
+	return output
+}
+
+func NewFrontendWorkspacePath(config *koanf.Koanf) *FrontendWorkspacePathConfig {
```
Comment (czezi) [inline workspace-service/app/config/frontend_worskpace_path.go:19]: Please name it frontend workspace config

File: workspace-service/app/utils/fs_utils.go [hunk 4]
```diff
@@ -17,8 +18,8 @@ func CheckIfWorkspaceExists(workspaceId string) (bool, error) {
 	return false, err
 }
 
-func CheckIfFrontendWorkspaceExists(storyHashId, workspaceId string) (bool, error) {
-	_, err := os.Stat("/workspaces/" + workspaceId + "/" + storyHashId)
+func CheckIfFrontendWorkspaceExists(path string) (bool, error) {
```
Comment (czezi) [inline workspace-service/app/utils/fs_utils.go:21]: make it generic CheckIfDirExists

File: server.go [hunk 5]
```diff
@@ -450,12 +461,20 @@ func main() {
...
+		project.GET("/design/stories", storiesController.GetDesignStoriesOfProject)
 
 		stories := api.Group("/stories", middleware.AuthenticateJWT())
 
 		stories.POST("", storiesController.CreateStory)
 		stories.POST("/", storiesController.CreateStory)
 
+		designStory := stories.Group("/design", middleware.AuthenticateJWT())
+
+		designStory.POST("", storiesController.CreateDesignStory)
```
Comment (czezi) [inline server.go:473]: Remove this.

File: Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (czezi) [reply Dockerfile:5]:
```
RUN groupadd -g 1000 coder
RUN useradd -u 1000 -g coder coder
RUN usermod -aG sudo coder
RUN echo 'coder ALL=NOPASSWD: ALL' >> /etc/sudoers
```

File: Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (czezi) [reply Dockerfile:80]: It's already set in the ide Dockerfile.

**Review by czezi — CHANGES_REQUESTED** [review]

File: app/utils/get_directory_structure.go
Comment (czezi) [inline app/utils/get_directory_structure.go:1]: Put this in fs_utils file.

File: app/workflow_executors/step_executors/impl/open_ai_next_js_code_generation_executor.go [hunk 6]
```diff
@@ -0,0 +1,581 @@
...
+		storyService:         storyService,
+		activityLogService:   activityLogService,
+		designReviewService:  designReviewService,
+		s3Service:            s3Service,
+		llmAPIKeyService:     llmAPIKeyService,
+	}
+}
+
+func (openAiCodeGenerator OpenAiNextJsCodeGenerator) Execute(step steps.GenerateCodeStep) error {
+	fmt.Printf("Executing GenerateCodeStep: %s\n", step.StepName())
```
Comment (czezi) [inline app/workflow_executors/step_executors/impl/open_ai_next_js_code_generation_executor.go:53]: Move to logger.

**Review by czezi — CHANGES_REQUESTED** [review]

File: workspace-service/app/config/config.go [hunk 7]
```diff
@@ -29,7 +29,7 @@ func LoadConfig() (config *koanf.Koanf, err error) {
 				},
 				"autoremove": "false",
 				"volume": map[string]interface{}{
-					"source": "supercoder_workspaces",
+					"source": "ai-developer_workspaces",
```
[the remark this answers is not in the document]
Comment (rounak610) [reply workspace-service/app/config/config.go]: fixed

File: workspace-service/app/config/frontend_worskpace_path.go
[the remark this answers is not in the document]
Comment (rounak610) [reply workspace-service/app/config/frontend_worskpace_path.go:1]: fixed

File: app/utils/get_directory_structure.go
Comment (rounak610) [reply app/utils/get_directory_structure.go:1]: but this function is used by next js server test executor and next js code generator

File: Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (rounak610) [reply Dockerfile:83]: fixed

**Review by czezi — APPROVED** [review]

**Pull request merged**